python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -2,11 +2,49 @@
 from __future__ import division, print_function, absolute_import
 
 import numpy as np
+from scipy.optimize import leastsq
 from scipy.spatial import cKDTree
 
 import lsst.pipe.base as pipeBase
 
 
+def _rotation_matrix_chi_sq(flattened_rot_matrix, src_pattern, ref_pattern):
+    """Compute the squared differences for least squares fitting.
+
+    Given a flattened rotation matrix, a N point pattern from the source
+    catalog and the reference pattern the sources match to, compute return
+    the squared differences between the points in the two patterns after
+    rotation.
+
+    Parameters
+    ----------
+    flattened_rot_matrix : float array
+        A flattened array representing a 3x3 rotation matrix.
+    src_patterm : float array of 3 vectors
+        A array containing N, 3 vectors
+    ref_pattern : float array of 3 vectors
+        A array containing N, 3 vectors
+
+    Returns
+    -------
+    float array
+        Array of differences between the points of the source pattern
+        rotated into the reference frame and the reference pattern rotated
+        into the source frame.
```

Comment:
I'm afraid I don't know what this means. What is the "difference between the points..."?

Also, why do you rotate the source into the reference frame *and also* rotate the reference pattern into the source frame. Naively I would expect only one rotation would be required to compare patterns of points in the two frames.